### ISSUE TITLE
robot-simulator: code && instructions is not matching

### ISSUE THREAD

**Issue by exklamationmark** [exercism/go#1159]
Hi,

I'm was trying to do the `robot-simulator` exercise. However, I found what was written in the README && what's in the test are wildly different.

While I appreciate a more complex & closer-to-life simulation, which was presented in the test,
it was not very clear what to do from the instructions alone.

What do you think about updating the exercise, such that:
- The tests and instruction match the [original specification](https://github.com/exercism/problem-specifications/tree/master/exercises/robot-simulator)
- Update the README && bonus test to add the extra conditions:
  - Bounded room with wall (instead of infinite grid)
  - Multiple robot being tested (instead of just one robot)

P.S: I also don't fully agree with some of the types && test defined, so I [made some changes](https://github.com/exercism/go/compare/master...exklamationmark:robot_simulator?expand=1)

**Comment by hilary** [comment]
I agree we should fix the problem of the description not matching well with the tests, but I'd like to think this through a bit before we launch into it.

How about if we make this problem a multi-step problem, changing the robot types as we go? I think it's fine to have them solve the easier problem first, then move on to the more difficult.

However... a fundamental tenant of exercism is that we try to avoid imposing any specific solution on the learner. I was concerned that the original version of the problem imposed too much structure. Your version imposes even more!

Can you come up with a way (try thinking in interfaces rather than in data structures) to test without imposing a solution? I'll take a thwack at it too 😈
